packages/core/src/base-axis-chart.ts

```diff
@@ -421,17 +422,21 @@ export class BaseAxisChart extends BaseChart {
 		}
 
 		yGrid.ticks(scales.y.numberOfTicks || Configuration.scales.y.numberOfTicks);
-
+		
 		const g = this.innerWrap.select(".y.grid")
 			.attr("transform", "translate(0, 0)")
 			.call(yGrid);
-
+			
 		this.cleanGrid(g);
 		this.fillTickThresholds(g);
 	}
 
 	fillTickThresholds(yGrid) {
 		const { thresholds } = this.options;
+		if (typeof thresholds === "undefined") {
+			return;
+		}
+
```

Comment:
This would probably be more effective going into the function above where you're calling `fillTickThresholds`

```js
if (this.options.thresholds) {
  this.fillTickThresholds(g);
}
```

Reply:
Also, we should add a few checks to make sure the `thresholds` config provided is formatted properly or does not contain bad values. If so, we can use `console.error` to let the developer know

Reply:
You said that we're gonna be provided a set of colors instead of specifying them in options.thresholds right? Will options.thresholds be a flat array of numbers then?

Reply:
This could be the format:
![image](https://user-images.githubusercontent.com/14989804/47050557-d716bf00-d16e-11e8-8389-01deb501c7b4.png)

Reply:
Although, I think style would make more sense than color

Reply:
Or type.

Reply:
Good point Z